### DISCUSSION TITLE
Data driven estimation for some of the backgrounds

### DISCUSSION THREAD

**Discussion by hephysicist** [columnflow/columnflow#411] [General]
Hi all! 
I want to bring up the discussion of creating additional datasets and corresponding histograms representing the backgrounds after the object selection and reduction of the initial dataset. 
For the Run2 CP analysis, the QCD background was estimated using the data-driven approach. The idea was to define 4 regions using two variables: relative sign of the tau leptons pair and muon isolation (ABCD method). Then one needs to estimate a transfer factor by finding a ratio between the number of events for data and all MC samples in two corresponding regions. Then we need to find a difference between the data histogram and all background histograms for each variable of interest. This is done for the control region. Finally, we scale the residual histogram from the previous step to the transfer factor. Currently, it is possible to select and save these datasets separately. However, I didn't find any way of performing all these manipulations over the histograms. 
Do you have any ideas on how this thing can be done?
Looking forward to any feedback.
Stepan.

**Comment by pkausw** [comment]
Hi @hephysicist ,

apologies for the late reply! This is of course an important feature for many analyses, so it's very nice that you opened this discussion :+1:

From the top of my head, I think the following might work:
- You should be able to use categories to implement the different reagions you describe, i.e. the control region and your ABCD regions. This should also resolut in separate histograms for these regions, which you can then use to calculate the transfer factors. Note: it might be necessary to add some of these histograms if your regions consist of smaller regions since columnflow builds the smallest units and then puts them together to create larger categories. See also the [category documentation](https://columnflow.readthedocs.io/en/latest/user_guide/building_blocks/categories.html)
- Currently, there is no task that calculates scale factors. However, you can also write your own law task to accomplish this, which will probably need to require the `CreateHistograms` task with a given set of categories and processes. Now that I think about it, it might also be able to write a producer that requires these tasks and then loads the output - I'm not sure how the histograms are saved and whether they are loadable as an awkward array, which iirc is currently the format that is supported by the ProduceColumns task. Maybe @mafrahm can comment on this since he implemented most of the histogram logic.

I assume that you've put in some work in the meanwhile, so it would also be very interesting to hear the solution that you came up with!
